tests/scenarios.go

```diff
@@ -431,11 +450,43 @@ func addParameterFrom(t gobdd.StepTest, ctx gobdd.Context, name string, path str
 	ctx.Set(requestArgsKey{}, append(GetRequestArguments(ctx), value))
 }
 
+func addPathArgumentWithValue(t gobdd.StepTest, ctx gobdd.Context, param string, value string) {
+	// Get request builder for the current scenario
+	request, err := getRequestFunction(ctx)
+	if err != nil {
+		t.Error(err)
+	}
+
+	in := make([]reflect.Value, request.Type().NumIn())
+	in[0] = reflect.ValueOf(GetCtx(ctx))
+	// The order of the path arguments in the scenario definition
+	// must match the order of the arguments in the function signature
+	// Here we keep track of which numbered path argument we're setting
+	pathCount, _ := ctx.Get(pathParamCountKey{})
+	varType := reflect.New(request.Type().In(pathCount.(int)))
+	ctx.Set(pathParamCountKey{}, pathCount.(int)+1)
+
+	templatedValue := Templated(t, GetData(ctx), value)
+
+	json.Unmarshal([]byte(templatedValue), varType.Interface())
+	GetRequestParameters(ctx)[param] = varType.Elem()
+
+	ctx.Set(requestArgsKey{}, append(GetRequestArguments(ctx), varType.Elem()))
+}
+
 func addParameterWithValue(t gobdd.StepTest, ctx gobdd.Context, param string, value string) {
 	// Get request builder for the current scenario
 	request, err := getRequestBuilder(ctx)
+
+	// If the getRequestBuilder method errors, its likely because we're trying to add a
+	// path param instead of a query param.
 	if err != nil {
-		t.Error(err)
+		if strings.Contains(err.Error(), "this is likely a path argument") {
+			addPathArgumentWithValue(t, ctx, param, value)
```

Comment:
Here we check instead if we get `request.Type().Kind() == Func`, then we call the helper `addPathArgumentWithValue` with `request` as an additional param. Let's also make it clear that this helper should not be a "step" function, like this one.